eZ/Publish/Core/Persistence/Cache/AbstractInMemoryHandler.php

```diff
@@ -0,0 +1,318 @@
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Persistence\Cache;
+
+use eZ\Publish\Core\Persistence\Cache\InMemory\InMemoryCache;
+use eZ\Publish\SPI\Persistence\Handler as PersistenceHandler;
+use Symfony\Component\Cache\Adapter\TagAwareAdapterInterface;
+
+/**
+ * Class AbstractInMemoryHandler.
+ *
+ * Abstract handler for use in other Persistence Cache Handlers.
+ */
+abstract class AbstractInMemoryHandler
+{
+    /**
+     * NOTE: $cache and $inMemory is private in order to abstract interactions across both,
+     *       and be able to optimize this later without affecting all classes extending this.
+     *
+     * @var \Symfony\Component\Cache\Adapter\TagAwareAdapterInterface
+     */
+    private $cache;
+
+    /**
+     * @var \eZ\Publish\SPI\Persistence\Handler
+     */
+    protected $persistenceHandler;
+
+    /**
+     * @var \eZ\Publish\Core\Persistence\Cache\PersistenceLogger
+     */
+    protected $logger;
+
+    /**
+     * @var \eZ\Publish\Core\Persistence\Cache\InMemory\InMemoryCache
+     */
+    private $inMemory;
+
+    /**
+     * Setups current handler with everything needed.
+     *
+     * @param \Symfony\Component\Cache\Adapter\TagAwareAdapterInterface $cache
+     * @param \eZ\Publish\SPI\Persistence\Handler $persistenceHandler
+     * @param \eZ\Publish\Core\Persistence\Cache\PersistenceLogger $logger
+     * @param \eZ\Publish\Core\Persistence\Cache\InMemory\InMemoryCache $inMemory
+     */
+    public function __construct(
+        TagAwareAdapterInterface $cache,
+        PersistenceHandler $persistenceHandler,
+        PersistenceLogger $logger,
+        InMemoryCache $inMemory
+    ) {
+        $this->cache = $cache;
+        $this->persistenceHandler = $persistenceHandler;
+        $this->logger = $logger;
+        $this->inMemory = $inMemory;
+    }
+
+    /**
+     * Load one cache item from cache and loggs the hits / misses.
+     *
+     * Load items from in-memory cache, symfony cache pool or backend in that order.
+     * If not cached the returned objects will be placed in cache.
+     *
+     * @param int|string $id
+     * @param string $keyPrefix E.g "ez-content-"
+     * @param callable $backendLoader Function for loading missing objects, gets array with missing id's as argument,
+     *                                expects return value to be array with id as key. Missing items should be missing.
+     * @param callable $cacheTagger Gets cache object as argument, return array of cache tags.
+     * @param callable $cacheIndexes Gets cache object as argument, return array of cache keys.
+     *
+     * @return object
+     */
+    final protected function getCacheValue(
+        $id,
+        string $keyPrefix,
+        callable $backendLoader,
+        callable $cacheTagger,
+        callable $cacheIndexes,
+        string $keyPostfix = ''
+    ) {
+        $key = $keyPrefix . $id . $keyPostfix;
```

Comment:
I'll need to check, usually it was needed in order to extract id, as arrays in those cases always return with id as key.

Reply:
NOTE: originally I did intend to drop `$keyPostfix ` at least, but it would mean change to several cache keys, so we can do it but that will be the effect.